include/evse_security/crypto/interface/crypto_types.hpp

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <optional>
+
+namespace evse_security {
+
+enum class CertificateValidationError {
+    NoError,
+    Expired,
+    InvalidSignature,
+    IssuerNotFound,
+    InvalidLeafSignature,
+    InvalidChain,
+    Unknown,
+};
+
+class CertificateLoadException : public std::runtime_error {
+public:
+    using std::runtime_error::runtime_error;
+};
+
+struct CryptoHandle {
+    virtual ~CryptoHandle() {
+    }
+    virtual void* get_raw_handle() = 0;
```

Comment:
Is this used anywhere? If not, consider removing it.

Reply:
Yes, it is the generic handle used by X509Handle. The KeyHandle is not used at the moment but it is required for the future TPM support.

Reply:
I meant the `get_raw_handle()` method. My IDE at least doesn't seem to find any usages of it, and you mentioned in a previous comment that the internal handles shouldn't really be accessed from the outside.

Reply:
Yes, that is correct, I think it was thought to have it there for some debug purposes, however I am not entirely sure that function is required.

Reply:
Alright, feel free to resolve this when you've done that.